.gitlab/pipeline.yml

```diff
@@ -10,56 +10,138 @@ include:
   - remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
 
 stages:
+  - spack_base
+  - spack_configs
   - build
   - test
 
 variables:
   SOURCE_DIR: /pika/source
   BUILD_DIR: /pika/build
+  SPACK_VERSION: v0.20.1
+
+.cmake_variables_common:
+  variables:
+    CMAKE_COMMON_FLAGS: "-DPIKA_WITH_COMPILER_WARNINGS=ON -DPIKA_WITH_COMPILER_WARNINGS_AS_ERRORS=ON -DPIKA_WITH_CHECK_MODULE_DEPENDENCIES=ON -DPIKA_WITH_EXAMPLES=ON -DPIKA_WITH_TESTS=ON"
+
+.variables_gcc11_config:
+  variables:
+    ARCH: linux-ubuntu22.04-broadwell
+    BOOST: boost@1.78.0
+    BUILD_TYPE: Debug
+    COMPILER: gcc@11.2.0
```

Comment:
Why not 11.4.0?

Reply:
Ok, but that doesn't mean we should downgrade the compiler. I can think of at least ways forward here:
- Check if the system compiler is the compiler we want to install through spack, and if it is we skip building the compiler through spack.
- Check if we can install the same version alongside the system compiler. I'm not quite sure what spack will think of this.
I think the second option would be interesting to know if it works and nice since then we can control exactly how the compiler is built, but this one is more work. I'd be happy with the first option. Finally, I'd also be happy to have this done in a follow-up PR. This is not a blocker, I just wanted to know why we weren't using the latest version in the 11.X series. If you prefer to do this separately, please open an issue for it.